questdb-rs/src/ingress/mod.rs

```diff
@@ -1337,6 +1348,21 @@ fn add_webpki_roots(root_store: &mut rustls::RootCertStore) {
     }));
 }
 
+#[cfg(feature = "tls-native-certs")]
+fn add_os_roots(root_store: &mut RootCertStore) -> Result<()> {
+    let os_certs = rustls_native_certs::load_native_certs().map_err(|io_err| {
+        error::fmt!(TlsError, "Could not OS native TLS certificates: {}", io_err)
+    })?;
+    for cert in os_certs {
+        root_store
+            .add(&rustls::Certificate(cert.0))
```

Comment:
`add()` contract says this:
```
    /// Add a single DER-encoded certificate to the store.
    ///
    /// This is suitable for a small set of root certificates that are expected to parse
    /// successfully. For large collections of roots (for example from a system store) it
    /// is expected that some of them might not be valid according to the rules rustls
    /// implements. As long as a relatively limited number of certificates are affected,
    /// this should not be a cause for concern. Use [`RootCertStore::add_parsable_certificates`]
    /// in order to add as many valid roots as possible and to understand how many certificates
    /// have been diagnosed as malformed.
```
This tells me not all certificates from the OS trust store can be added and failing to add a cert is not necessarily a critical error. OTOH: Just silently ignoring it does would lead to hard to troubleshoot issues so that's not great either. WDYT?

Reply:
thinking out-loud: maybe just logging it at some lower level (DEBUG or similar) is the best thing to do here? It would allow for troubleshooting yet it would not fail clients to start in the case OS store has certificates which rustls dislikes for some reason.

Reply:
This is what `reqwest` (very treaded code) does: https://github.com/seanmonstar/reqwest/blob/6792f697fcdb27c47dcbf7bd05f23368d1d4ac80/src/async_impl/client.rs#L485

Reply:
I should probably do the same. Generally, I don't think this code should log: It's just a library.

Reply:
the reqwest strategy looks sensibe: they fail to start only if they manage to load no certificate.

Reply:
I've updated the code now. It does the same, but by calling the `add_parsable_certificates` fn.
No logging on error though.

Reply:
looks good to me!